src/main.rs

```diff
@@ -490,18 +549,26 @@ fn run_cli(cli: Cli) -> Result<()> {
                         (&RecordData::default()).into()
                     };
 
-                    let row = missing.insert_and_ref(&data, &remote_id)?;
-                    (row, data)
+                    let row = undefined.insert_and_ref(&data, &remote_id)?;
+                    (row, data, remote_id)
+                }
+                RemoteIdState::Null(_, _) => {
+                    unreachable!(
+                        "A local record should not be present in the `NullRecords` table."
+                    );
+                }
```

Comment:
This is not logically unreachable code (as the database could be externally modified), so `unreachable!` shouldn't be used here. Maybe `bail!`?

Reply:
Should we assume that the database could be externally modified in the `db.rs` code as well? If e.g. the binary data is invalid, the code will also panic.

Reply:
Yeah, maybe we should handle data corruption more gracefully.

I think that it's not that bad to `panic!` in the above; just that `unreachable!` is semantically inaccurate.

Reply:
Makes sense; well, at least in this spot, recovering is quite sensible. In the binary data, even returning an error feels dangerous because if the database is malformed we pretty much want to bail as fast as possible and ask the user to try to fix it / figure out what is wrong using the `autobib util check` command. Because pretty much all of the database operations are undefined once the database is in an undefined state.